nixos/modules/services/web-apps/librespeed.nix

```diff
@@ -0,0 +1,452 @@
+{
+  config,
+  lib,
+  options,
+  pkgs,
+  ...
+}:
+let
+  cfg = config.services.librespeed;
+  opt = options.services.librespeed;
+
+  mapValue =
+    arg:
+    if (lib.isBool arg) then
+      lib.boolToString arg
+    else if ((lib.isInt arg) || (lib.isFloat arg)) then
+      toString arg
+    else
+      "\"${lib.escape [ "\"" ] (toString arg)}\"";
+
+  configFile = lib.pipe cfg.settings [
+    (lib.filterAttrs (_: val: val != null))
+    (lib.mapAttrs (name: val: "${name}=${mapValue val}"))
+    lib.attrValues
+    lib.concatLines
+    (pkgs.writeText "${cfg.package.name}-config.toml")
+  ];
+
+  librespeedAssets =
+    pkgs.runCommand "librespeed-assets"
+      {
+        serversList = ''
+          function get_servers() {
+            return ${builtins.toJSON cfg.frontend.servers}
+          }
+          function override_settings () {
+          ${lib.pipe cfg.frontend.settings [
+            (lib.mapAttrs (name: val: "  s.setParameter(\"${lib.escape [ "\"" ] name}\",${mapValue val});"))
+            lib.attrValues
+            lib.concatLines
+          ]}
+          }
+        '';
+      }
+      ''
+        cp -r ${pkgs.librespeed-rust}/assets $out
+        chmod 666 $out/servers_list.js
+        cat >$out/servers_list.js <<<"$serversList"
+        substitute ${pkgs.librespeed-rust}/assets/index.html $out/index.html \
+          --replace-fail "s.setParameter(\"telemetry_level\",\"basic\"); //enable telemetry" "override_settings();" \
+          --replace-fail "LibreSpeed Example" ${lib.escapeShellArg (lib.escapeXML cfg.frontend.pageTitle)} \
+          --replace-fail "PUT@YOUR_EMAIL.HERE" ${lib.escapeShellArg (lib.escapeXML cfg.frontend.contactEmail)} \
+          --replace-fail "TO BE FILLED BY DEVELOPER" ${lib.escapeShellArg (lib.escapeXML cfg.frontend.contactEmail)}
+      '';
+in
+{
+  options.services.librespeed = {
+    enable = lib.mkEnableOption "LibreSpeed server";
+    package = lib.mkPackageOption pkgs "librespeed-rust" { };
+
+    domain = lib.mkOption {
+      description = ''
+        If not `null`, this will add an entry to `services.librespeed.servers` and
+        configure librespeed to use TLS.
+      '';
+      default = null;
+      type = with lib.types; nullOr nonEmptyStr;
+    };
+
+    downloadIPDB = lib.mkOption {
+      description = ''
+        Whether to download the IP info database before starting librespeed.
+        Disable this if you want to use the Go implementation.
+      '';
+      default = !(cfg.secrets ? "ipinfo_api_key");
+      defaultText = lib.literalExpression ''!(${opt.secrets} ? "ipinfo_api_key")'';
```

Comment:
I think several of these `literalExpression`s using `opt` should be prefixed with `config.`

<img width="1846" height="488" alt="image" src="https://github.com/user-attachments/assets/2e2d7d3b-75b7-4539-a854-2d95305580f0" />


```suggestion
      defaultText = lib.literalExpression ''!(config.${opt.secrets} ? "ipinfo_api_key")'';
```